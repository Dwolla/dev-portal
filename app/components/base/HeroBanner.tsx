import styled from "@emotion/styled";
import {
  AlgoliaAutocomplete,
  AlgoliaAutocompleteProps,
} from "algolia-search-components";
import algoliasearch from "algoliasearch/lite";
import React from "react";
import { styled as muiStyled } from "@mui/material/styles";
import { POPPINS, ROBOTO } from "../typography";
import { WHITE_070, WHITE_PRIMARY } from "../colors";
import { breakDown } from "../breakpoints";

// Styled Banner component
const StyledBanner = styled.div`
  height: 412px;
  width: 100%;
  padding: 40px;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
  border-radius: 16px;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const BannerContentWrap = styled.div`
  max-width: 525px;
  display: flex;
  flex-direction: column;
  align-items: center;

  > *:last-child {
    margin-bottom: 0;
  }
`;

const StyledTopic = styled.div`
  color: ${WHITE_PRIMARY};
  font-family: ${POPPINS};
  font-style: normal;
  font-weight: 500;
  font-size: 40px;
  line-height: 123.5%;
  text-align: center;
  margin-bottom: 32px;
  letter-spacing: 0.25px;

  @media (${breakDown("sm")}) {
    font-size: 30px;
  }

  @media (${breakDown("xs")}) {
    font-size: 22px;
  }
`;

const StyledDescription = styled.div`
  font-family: ${ROBOTO};
  font-style: normal;
  font-weight: 300;
  font-size: 16px;
  line-height: 175%;
  letter-spacing: 0.15px;
  text-align: center;
  margin-bottom: 24px;
  color: ${WHITE_070};

  @media (${breakDown("sm")}) {
    font-size: 12px;
  }
`;

const AlgoliaSearchWrapper = muiStyled("div", { name: "SearchWrapper" })({
  display: "inline-block",
  verticalAlign: "middle",
  width: "100%",
});

const algoliaSearch: AlgoliaAutocompleteProps = {
  branch: "main",
  searchClient: algoliasearch("L2PPGO4SBB", "6a6c05b578da5aa729df7f53776e9f76"),
  siteId: "e19df9e6-7024-443d-8ec0-26e8312ce0f9",
  searchOptions: { hitsPerPage: 5 },
};

type ImageProps = {
  src: string;
  alt: string;
};

type Props = {
  heroImage: ImageProps;
  topic: string;
  description: string;
  searchBar?: boolean;
};

function HeroBanner({ heroImage, topic, description, searchBar }: Props) {
  return (
    <StyledBanner
      style={{
        backgroundImage: `url(${heroImage.src})`,
        backgroundRepeat: "no-repeat",
        backgroundSize: "cover",
        backgroundPosition: "center",
      }}
    >
      <BannerContentWrap>
        <StyledTopic>{topic}</StyledTopic>
        <StyledDescription>{description}</StyledDescription>
        {searchBar && (
          <AlgoliaSearchWrapper>
            <AlgoliaAutocomplete {...algoliaSearch} />
          </AlgoliaSearchWrapper>
        )}
      </BannerContentWrap>
    </StyledBanner>
  );
}

export default HeroBanner;
