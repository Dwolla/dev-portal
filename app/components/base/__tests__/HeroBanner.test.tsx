import renderer from "react-test-renderer";
import HeroBanner from "../HeroBanner";
import hero from "../../../assets/images/content-images/hero-image-home.svg";

test("Default HeroBanner", () => {
  const tree = renderer
    .create(
      <HeroBanner
        heroImage={{
          src: hero,
          alt: "Some alt text",
        }}
        topic="Build with Dwolla"
        description="Documentation, tools and resources for developers to integrate Dwolla's account-to-account payments Platform."
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("HeroBanner - with SearchBar", () => {
  const tree = renderer
    .create(
      <HeroBanner
        heroImage={{
          src: hero,
          alt: "Some alt text",
        }}
        topic="Build with Dwolla"
        description="Documentation, tools and resources for developers to integrate Dwolla's account-to-account payments Platform."
        searchBar
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
