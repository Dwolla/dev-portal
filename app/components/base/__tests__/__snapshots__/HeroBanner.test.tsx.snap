// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default HeroBanner 1`] = `
<div
  className="css-s65f2z-StyledBanner e38sxrk3"
  style={
    {
      "backgroundImage": "url(test-image-stub)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="css-l0a4nd-BannerContentWrap e38sxrk2"
  >
    <div
      className="css-1ubtxoh-StyledTopic e38sxrk1"
    >
      Build with Dwolla
    </div>
    <div
      className="css-1oef8wh-StyledDescription e38sxrk0"
    >
      Documentation, tools and resources for developers to integrate Dwolla's account-to-account payments Platform.
    </div>
  </div>
</div>
`;

exports[`HeroBanner - with SearchBar 1`] = `
<div
  className="css-s65f2z-StyledBanner e38sxrk3"
  style={
    {
      "backgroundImage": "url(test-image-stub)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="css-l0a4nd-BannerContentWrap e38sxrk2"
  >
    <div
      className="css-1ubtxoh-StyledTopic e38sxrk1"
    >
      Build with Dwolla
    </div>
    <div
      className="css-1oef8wh-StyledDescription e38sxrk0"
    >
      Documentation, tools and resources for developers to integrate Dwolla's account-to-account payments Platform.
    </div>
    <div
      className="css-4467i6-SearchWrapper-root"
    >
      <div />
    </div>
  </div>
</div>
`;
